ngraph/core/src/op/dft.cpp

```diff
@@ -0,0 +1,196 @@
+
+    element::Type axes_et = get_input_element_type(1);
+    NODE_VALIDATION_CHECK(this,
+                          axes_et == element::i64 || axes_et == element::i32 ||
+                              axes_et == element::u32 || axes_et == element::u64,
+                          "Axes element type must be i32, i64, u32 or u64");
+
+    PartialShape input_shape = PartialShape(get_input_partial_shape(0));
+    PartialShape axes_shape = PartialShape(get_input_partial_shape(1));
+    PartialShape output_shape = input_shape;
+
+    if (input_shape.rank().is_dynamic() || axes_shape.rank().is_dynamic() ||
+        !is_type<op::Constant>(input_value(1).get_node()))
+    {
+        set_output_type(0, get_input_element_type(0), output_shape);
+        return;
+    }
+
+    const auto input_rank = input_shape.rank().get_length();
+
+    NODE_VALIDATION_CHECK(this,
+                          axes_shape.rank().get_length() == 1,
+                          "Axes input must be 1D tensor. Got axes input rank: ",
+                          axes_shape.rank().get_length());
+
+    auto last_dim_with_two = input_shape[input_rank - 1] & Dimension(2);
+    NODE_VALIDATION_CHECK(this,
+                          !last_dim_with_two.get_interval().empty(),
+                          "The last dimension of input data must be 2. Got: ",
+                          input_shape[input_rank - 1]);
+
+    const auto& const_axes = get_constant_from_source(input_value(1));
+    auto axes = const_axes->cast_vector<int64_t>();
+
+    for (int64_t& axis : axes)
+    {
+        if (axis < 0)
+        {
+            axis += input_rank - 1;
+        }
+    }
+
+    NODE_VALIDATION_CHECK(this,
+                          input_rank >= axes.size() + 1,
+                          "The input rank must be greater than number of DFT axes. Got "
+                          "input rank: ",
+                          input_rank,
+                          ", number of axes: ",
+                          axes.size());
+
+    AxisVector axes_vector;
+    AxisSet axes_set;
+    for (const int64_t axis : axes)
+    {
+        axes_vector.push_back(static_cast<size_t>(axis));
+        axes_set.insert(static_cast<size_t>(axis));
+    }
+
+    NODE_VALIDATION_CHECK(
+        this, axes_vector.size() == axes.size(), "DFT axes must be unique. Got: ", axes_vector);
+
+    NODE_VALIDATION_CHECK(this,
+                          std::find(axes.begin(), axes.end(), input_rank - 1) == axes.end(),
+                          "FFT axis cannot be the last axis. Got axes: ",
+                          axes_vector);
+
+    if (input_values().size() == 2)
+    {
+        set_output_type(0, get_input_element_type(0), output_shape);
+        return;
+    }
+
+    element::Type signal_sizes_et = get_input_element_type(2);
+    NODE_VALIDATION_CHECK(this,
+                          signal_sizes_et == element::i64 || signal_sizes_et == element::i32 ||
```

Comment:
Maybe such checks can be replaced by signal_sizes_et.is_integral_number()?

Reply:
`signal_sizes_et.is_integral_number()` will return `true` for `u8`, `u16`, `i8`, `i16` too. But, according to the specification of DFT,  such element type are not supported for `signal_size` and `axes` inputs.